pkg/authorization/admission/restrictusers/subjectchecker.go

```diff
@@ -167,7 +154,7 @@ func (checker UserSubjectChecker) Allowed(subject kapi.ObjectReference, ctx *Rol
 	}
 
 	// System users can match only by name.
-	if subject.Kind != authorizationapi.UserKind {
+	if subject.Kind != rbac.UserKind {
```

Comment:
The entire `if` block and comment should be deleted.  Its purpose was to skip checking groups and selectors for system users, but now it is redundant.